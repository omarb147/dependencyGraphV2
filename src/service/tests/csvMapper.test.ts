import exampleMondayCSV from './csvMapperMock';
import {
  mapCSVtoObject,
  formatUserStory,
  getHeadersIndex,
} from '../csvMapper';

describe('mapCSVtoObject', () => {
  it('should return an array of row objects - Full version', () => {
    expect(mapCSVtoObject(exampleMondayCSV)).toMatchSnapshot();
  });
  it('should return an array of row objects - Simplified version', () => {
    const simpleCSV = `Prod PR template: Production Deploy YYYY-MM-DD {NUM_DEPLOY_TODAY}",,,,,,,
    ,,,,,,,
    "Phase 3, Sprint 1: AACKU, on the home page, when I click on my contact database button, I am taken to the contact database of the latest survey",,,,,,,
    Name,Status,Labels,Points,Person,Item ID,Priority Order,Item ID (auto generated)
    "SPRINT GOAL: AACKU, on the home page, when I click on my contact database button, I am taken to the contact database of the latest survey",Access Latest Survey,SprintGoal,,,504135682,,504135682
    "(2) - AACKUser on the home page, when there are no surveys in an organisation, the button ""My Contact Database"" is disabled",Access Latest Survey,SprintGoal,2,,505730979,,505730979
    "[TIMEBOX 1/2day*1dev] AACRC/CKUser on the CDB page, I can see a dropdown for the CC/PC/I and status column for existing surveys and surveys created with existing products (e.g. TRR product)",Go Live Support,SprintReady,,Rob Cronin,507017759,,507017759
    Coaching Elio<>Mike,Coaching,SprintTemplate,,,507679478,,507679478
    `;
    expect(mapCSVtoObject(simpleCSV)).toMatchObject({
      tickets: [
        {
          itemId: '504135682',
          name: 'SPRINT GOAL: AACKU, on the home page, when I click on my contact database button, I am taken to the contact database of the latest survey',
          labels: 'SprintGoal',
          person: '',
          points: '',
          status: 'Access Latest Survey',
        },
        {
          itemId: '505730979',
          name: 'AACKUser on the home page, when there are no surveys in an organisation, the button "My Contact Database" is disabled',
          labels: 'SprintGoal',
          person: '',
          points: '2',
          status: 'Access Latest Survey',
        },
        {
          itemId: '507017759',
          labels: 'SprintReady',
          person: 'Rob Cronin',
          name: '[TIMEBOX 1/2day*1dev] AACRC/CKUser on the CDB page, I can see a dropdown for the CC/PC/I and status column for existing surveys and surveys created with existing products (e.g. TRR product)',
          points: '',
          status: 'Go Live Support',
        },
        {
          itemId: '507679478',
          labels: 'SprintTemplate',
          person: '',
          points: '',
          name: 'Coaching Elio<>Mike',
          status: 'Coaching',
        },
      ],
      headers: [
        {
          color: 'green',
          id: 'accessLatestSurvey',
          name: 'Access Latest Survey',
        },
        {
          color: 'green',
          id: 'goLiveSupport',
          name: 'Go Live Support',
        },
        {
          color: 'green',
          id: 'coaching',
          name: 'Coaching',
        },
      ],
    });
  });
});

describe('formatUserStory', () => {
  it('should return user story without point and quotation mark', () => {
    const text = '"(2) - AACKUser on the home page, when there are no surveys in an organisation, the button ""My Contact Database"" is disabled';
    expect(formatUserStory(text)).toEqual('AACKUser on the home page, when there are no surveys in an organisation, the button "My Contact Database" is disabled');
  });

  it('should return user story with square bracket but without the quotation marks', () => {
    const text = '"[TIMEBOX 1/2day*1dev] AACRC/CKUser on the CDB page, I can see a dropdown for the CC/PC/I and status column for existing surveys and surveys created with existing products (e.g. TRR product)"';
    expect(formatUserStory(text)).toEqual('[TIMEBOX 1/2day*1dev] AACRC/CKUser on the CDB page, I can see a dropdown for the CC/PC/I and status column for existing surveys and surveys created with existing products (e.g. TRR product)');
  });

  it('should return user story as is if already formatted correctly', () => {
    const text = 'Coaching Roblex';
    expect(formatUserStory(text)).toEqual('Coaching Roblex');
  });
});

describe('getHeadersIndex', () => {
  it('should return the index of the header row in the csv', () => {
    const csv = exampleMondayCSV.split('\n');
    expect(getHeadersIndex(csv)).toEqual(5);
    expect(csv[5]).toEqual(expect.stringMatching('Name'));
    expect(csv[5]).toEqual(expect.stringMatching('Status'));
    expect(csv[5]).toEqual(expect.stringMatching('Label'));
    expect(csv[5]).toEqual(expect.stringMatching('Points'));
    expect(csv[5]).toEqual(expect.stringMatching('Item ID'));
    expect(csv[5]).toEqual(expect.stringMatching('Priority Order'));
  });
  it('should return undefined if it cannot find the header row in the csv', () => {
    expect(getHeadersIndex([])).toEqual(undefined);
  });
});
