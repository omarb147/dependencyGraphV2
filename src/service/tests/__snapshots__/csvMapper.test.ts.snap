// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapCSVtoObject should return an array of row objects - Full version 1`] = `
Array [
  Object {
    "itemId": "504135682",
    "labels": "SprintGoal",
    "name": "SPRINT GOAL: AACKU, on the home page, when I click on my contact database button, I am taken to the contact database of the latest survey",
    "person": "",
    "points": "",
    "status": "Access Latest Survey",
  },
  Object {
    "itemId": "505730979",
    "labels": "SprintGoal",
    "name": "AACKUser on the home page, when there are no surveys in an organisation, the button \\"My Contact Database\\" is disabled",
    "person": "",
    "points": "2",
    "status": "Access Latest Survey",
  },
  Object {
    "itemId": "504135687",
    "labels": "",
    "name": "AACRCUser on the /survey/create page, when I create a survey, I can choose an existing survey from a dropdown to copy from",
    "person": "",
    "points": "3",
    "status": "Create Repeat Survey",
  },
  Object {
    "itemId": "507679466",
    "labels": "SprintTemplate",
    "name": "[TIMEBOX 1h*1dev] Backlog Refinement",
    "person": "",
    "points": "1",
    "status": "Sprint Meeting",
  },
  Object {
    "itemId": "507679467",
    "labels": "SprintTemplate",
    "name": "[TIMEBOX 1h*2dev] Technical Refinement",
    "person": "",
    "points": "2",
    "status": "Sprint Meeting",
  },
  Object {
    "itemId": "507679474",
    "labels": "SprintTemplate",
    "name": "[TIMEBOX 1h*2dev] Technical Refinement",
    "person": "",
    "points": "2",
    "status": "Sprint Meeting",
  },
  Object {
    "itemId": "507679465",
    "labels": "SprintTemplate",
    "name": "[TIMEBOX 30m] Clean up of any new sentry errors",
    "person": "",
    "points": "0.5",
    "status": "Tech Debt",
  },
  Object {
    "itemId": "507679486",
    "labels": "SprintTemplate",
    "name": "[TIMEBOX 1hr*2dev] Tech ref preparation",
    "person": "",
    "points": "2",
    "status": "Continuous Improvement",
  },
  Object {
    "itemId": "507679468",
    "labels": "SprintTemplate",
    "name": "Coaching Rob<>Alex",
    "person": "",
    "points": "1",
    "status": "Coaching",
  },
  Object {
    "itemId": "507679472",
    "labels": "SprintTemplate",
    "name": "Coaching Omar<>Mike",
    "person": "",
    "points": "1",
    "status": "Coaching",
  },
  Object {
    "itemId": "507679478",
    "labels": "SprintTemplate",
    "name": "Coaching Elio<>Mike",
    "person": "",
    "points": "",
    "status": "Coaching",
  },
  Object {
    "itemId": "507679484",
    "labels": "SprintTemplate",
    "name": "Coaching Frank<>Mike",
    "person": "",
    "points": "",
    "status": "Coaching",
  },
  Object {
    "itemId": "507017759",
    "labels": "SprintReady",
    "name": "[TIMEBOX 1/2day*1dev] AACRC/CKUser on the CDB page, I can see a dropdown for the CC/PC/I and status column for existing surveys and surveys created with existing products (e.g. TRR product)",
    "person": "Rob Cronin",
    "points": "",
    "status": "Go Live Support",
  },
  Object {
    "itemId": "504827399",
    "labels": "SprintReady",
    "name": "AACRCUser on the /contact-database page, when I look at the left side of the \\"Approve selected\\" button, I see a button labelled \\"Freeze contacts\\"",
    "person": "",
    "points": "",
    "status": "Transfer Previous Contacts",
  },
  Object {
    "itemId": "504786209",
    "labels": "SprintReady",
    "name": "AACRCUser on the /survey/create page, when I click on an existing survey from the dropdown, the \\"from email address\\" is pre-populated from the chosen survey",
    "person": "",
    "points": "",
    "status": "Create Repeat Survey",
  },
  Object {
    "itemId": "504779210",
    "labels": "SprintReady",
    "name": "AACRCUser on the /survey/create page, when I click on the choose from existing survey toggle, it shows me the dropdown for the existing surveys",
    "person": "",
    "points": "",
    "status": "Create Repeat Survey",
  },
  Object {
    "itemId": "504785148",
    "labels": "Blocked",
    "name": "AADev on the /survey/create page, when I click on the \\"next\\" button to create a survey, the survey is created with the same settings of the copied survey",
    "person": "",
    "points": "",
    "status": "Create Repeat Survey",
  },
  Object {
    "itemId": "504792052",
    "labels": "SprintReady",
    "name": "AACRCUser on the /survey/create page, when I click on the \\"next\\" button, I will be taken to the /unit-confirmation page",
    "person": "",
    "points": "",
    "status": "Create Repeat Survey",
  },
  Object {
    "itemId": "504836657",
    "labels": "",
    "name": "AADev on the /contact-database page, when I click on the \\"Freeze contacts\\" button, it will create a contact snapshot of the contacts in the database",
    "person": "",
    "points": "",
    "status": "Transfer Previous Contacts",
  },
  Object {
    "itemId": "504812527",
    "labels": "",
    "name": "AACRCUser, on the /contact-database page, when I click the \\"Freeze contacts\\" button in the top right, I see a success message saying \\"Survey contacts frozen\\"",
    "person": "",
    "points": "",
    "status": "Transfer Previous Contacts",
  },
  Object {
    "itemId": "504829900",
    "labels": "",
    "name": "AACRCUser, on the delivery platform, when I create a survey based on the new survey flow, it will send an email to approved and active status contacts on the frozen contact database",
    "person": "",
    "points": "",
    "status": "Transfer Previous Contacts",
  },
  Object {
    "itemId": "505221317",
    "labels": "",
    "name": "AADev, when I run the import script for the new hub CSV format, I can see the simple fields appear on the CDB page",
    "person": "",
    "points": "",
    "status": "Go Live Support",
  },
  Object {
    "itemId": "505222629",
    "labels": "",
    "name": "AADev, when I run the import script for the new hub CSV format, I can see the CC/PC fields appear on the CDB page",
    "person": "",
    "points": "",
    "status": "Go Live Support",
  },
  Object {
    "itemId": "505786165",
    "labels": "",
    "name": "AACRCUser on the surveyCreatePage when I toggle the copy from previous survey button and re-toggle it, the from email address is now blank",
    "person": "",
    "points": "",
    "status": "Create Repeat Survey",
  },
  Object {
    "itemId": "507108734",
    "labels": "",
    "name": "AADev, when I call the POST of /contactsnapshot, it will create snapshot of the contacts",
    "person": "",
    "points": "",
    "status": "Transfer Previous Contacts",
  },
  Object {
    "itemId": "507679489",
    "labels": "SprintTemplate",
    "name": "Coaching Omar <>Mike M/Josh",
    "person": "",
    "points": "",
    "status": "Coaching",
  },
  Object {
    "itemId": "508516460",
    "labels": "",
    "name": "Sprint 2 Tickets ~~~~~~~~~",
    "person": "",
    "points": "",
    "status": "",
  },
  Object {
    "itemId": "507734065",
    "labels": "Removed",
    "name": "[TIMEBOX 1hr*2.5dev]Kaizen Starter",
    "person": "\\"Alex Wong, Omar Bello, Mike Riddelsdell, Elio Stalteri, Frank Shenton, Rob Cronin, Tom Elliott\\"",
    "points": "",
    "status": "Additional Meeting",
  },
  Object {
    "itemId": "508516508",
    "labels": "",
    "name": "[TIMEBOX] AACRCUser, on the unit confirmation page, I can edit a unit combo and save/approve without an error",
    "person": "",
    "points": "",
    "status": "Bug",
  },
  Object {
    "itemId": "511383991",
    "labels": "LastSprint",
    "name": "AAUser, I can see all options in all lists in alphabetical order",
    "person": "\\"Omar Bello, Frank Shenton\\"",
    "points": "5",
    "status": "UX/UI",
  },
  Object {
    "itemId": "511619117",
    "labels": "",
    "name": "AACKUser, when I try and log on to the delivery platform using internet explorer, I am directed to a page that explains I cannot use this browser and highlights which browsers I can use",
    "person": "",
    "points": "",
    "status": "Go Live Support",
  },
  Object {
    "itemId": "511732749",
    "labels": "",
    "name": "AASuperAdmin, I can give myself permission to view a new contact database",
    "person": "",
    "points": "",
    "status": "Bug",
  },
]
`;
